### ISSUE TITLE
Go To Source Is Not Working

### ISSUE THREAD

**Issue by Alois-xx** [Microsoft/perfview#571]
Symbol lookup for our IIS hosted Symbol Server is not working. 

\perfview\src\TraceEvent\Symbols\SymbolReader.cs

A fix is to include a valid User Agent string
```
                           m_log.WriteLine("FindSymbolFilePath: In task, sending HTTP request {0}", fullUri);

                            var httpClient = new HttpClient();

                            // Add a user-agent header
                            var headers = httpClient.DefaultRequestHeaders;
                            // Use Chrome User Agent strings
                            headers.UserAgent.Add(new System.Net.Http.Headers.ProductInfoHeaderValue("Mozilla", "5.0"));
                            headers.UserAgent.Add(new System.Net.Http.Headers.ProductInfoHeaderValue("AppleWebKit", "537.36"));
                            headers.UserAgent.Add(new System.Net.Http.Headers.ProductInfoHeaderValue("Chrome", "41.0.2228.0"));
                            headers.UserAgent.Add(new System.Net.Http.Headers.ProductInfoHeaderValue("Safari", "537.36"));
                            var responseTask = httpClient.GetAsync(fullUri);
                            responseTask.Wait();
                            var response = responseTask.Result;

                            alive = true;
                            if (!canceled)
                            {
                                if (contentTypeFilter != null && !contentTypeFilter(response.Content.Headers.ContentType.ToString()))
                                    throw new InvalidOperationException("Bad File Content type " + response.Content.Headers.ContentType.ToString() + " for " + fullDestPath);

                                using (var fromStream = response.Content.ReadAsStreamAsync().Result)
                                {
                                    if (CopyStreamToFile(fromStream, fullUri, fullDestPath, ref canceled) == 0)
                                    {
                                        File.Delete(fullDestPath);
                                        throw new InvalidOperationException("Illegal Zero sized file " + fullDestPath);
                                    }
                                }
                                successful = true;
                            }
```
With that change it works like a charm for source server indexed pdbs from our symbol server. I needed to switch over to HttpClient because the HttpRequest user agent property does not exist on .NET Core.

**Comment by vancem** [comment]
Interesting.   When the source server stuff was first put in we had a user-agent header 
```
                        var fullUri = serverPath + "/" + pdbIndexPath.Replace('\\', '/');
                        try
                        {
                            var req = (System.Net.HttpWebRequest)System.Net.HttpWebRequest.Create(fullUri);
                            req.UserAgent = "Microsoft-Symbol-Server/6.13.0009.1140";
                            var response = req.GetResponse();
```
On 12/1/2017 we pulled this because it did not *seem* to be needed and was causing grief getting things to compile on older NetStandard platforms.   

I just tried some experiments and things seem to work for me, but details mater here.   First, what server are you using?    Is this http://msdl.microsoft.com/download/symbols or something else (sounds like it is something else).  I am also assuming that it is the PDB that is failing to be fetched, which should affect even method names, not just source code lookup (Fetching the source code itself I believe is done via a different mechanism).   Am I getting that right?   

It would be useful to try the user agent we used above.  I suspect that the server only cares that there IS a user agent, not which particular one.   

If adding a user agent back will fix things I am OK with doing that, (but I have to see what issues it causes compiling on old netstandard).

**Comment by Alois-xx** [comment]
According to the https://xxx/symbols page the Symbol Server is running on IIS. I can try this user agent string tomorrow to see if it works.

**Comment by Alois-xx** [comment]
@vancem: I have just tried out the "old" user agent string
`headers.UserAgent.Add(new System.Net.Http.Headers.ProductInfoHeaderValue("Microsoft-Symbol-Server", "6.13.0009.1140"));
`
that one does work. 
The symbol server proxy runs with these settings according to its web page:

SymbolServerProxy | SymbolServerProxy, Version=3.1.0.0, Culture=neutral, PublicKeyToken=null
-- | --
SymbolServerProxy Version | 3.0.1708.1003
IIS Version | Microsoft-IIS/8.5
.Net Runtime | 4.0.30319.42000

**Comment by vancem** [comment]
Fixed by #577
